### PR TITLE
 For #23970 - Migrate homescreen recent visits header to compose

### LINKED ISSUE #23968: Migrate RecentTabViewHolder to Compose
Breakage of https://github.com/mozilla-mobile/fenix/issues/22755

Refers to just the [RecentTabViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabsHeaderViewHolder.kt) which adds a "Jump back in" header.
[RecentTabViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt) which shows the recent tabs is already based on the compose framework.

The compose version can be added to the current RecyclerView but before doing so we should ensure there will be no performance regression introduced.
To profile the differences the steps from https://github.com/mozilla-mobile/fenix/issues/21854#issuecomment-1049392266 can be used.

### LINKED ISSUE #23969: Migrate RecentBookmarksHeaderViewHolder to Compose
Breakage of https://github.com/mozilla-mobile/fenix/issues/22755

Refers to just the [RecentBookmarksHeaderViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksHeaderViewHolder.kt) which adds a "Recent bookmarks" header.
[RecentBookmarksViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksViewHolder.kt) which shows the recent bookmarks is already based on the compose framework.

The compose version can be added to the current RecyclerView but before doing so we should ensure there will be no performance regression introduced.
To profile the differences the steps from https://github.com/mozilla-mobile/fenix/issues/21854#issuecomment-1049392266 can be used.

### LINKED ISSUE #23970: Migrate RecentVisitsHeaderViewHolder to Compose
Breakage of https://github.com/mozilla-mobile/fenix/issues/22755

Refers to just the [RecentVisitsHeaderViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentVisitsHeaderViewHolder.kt) which adds a "Recently visited" header.
[RecentlyVisitedViewHolder](https://github.com/mozilla-mobile/fenix/blob/f7d317fb4ea76428d4cc9578b76bd9ef6edf8ca0/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentlyVisitedViewHolder.kt)) which shows a scrollable list of the recent history items/groups is already based on the compose framework.

The compose version can be added to the current RecyclerView but before doing so we should ensure there will be no performance regression introduced.
To profile the differences the steps from https://github.com/mozilla-mobile/fenix/issues/21854#issuecomment-1049392266 can be used.

### PR DESCRIPTION
**Pull request by sarah541**
Fixes #23970 
Fixes #23969 
Fixes #23968 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentVisitsHeaderViewHolder.kt [hunk 1]
```diff
@@ -17,18 +39,79 @@ import org.mozilla.fenix.utils.view.ViewHolder
...
+                fontSize = 14.sp
+            ),
+            onClick = { onShowAllClick() }
+        )
+    }
+}
+
+@Composable
+@Preview
+private fun RecentVisitsHeaderViewHolderPreview() {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentVisitsHeaderViewHolder.kt:111]:
```suggestion
private fun RecentVisitsHeaderPreview() {
```

File: .experimenter.json [hunk 2]
```diff
@@ -10,13 +10,21 @@
       }
     }
   },
-  "default-browser-message": {
-    "description": "A small feature allowing experiments on the placement of a default browser message.",
+  "nimbus-validation": {
```
Comment (gabrielluong) [inline .experimenter.json:13]: Remove this from the commit.

**Review by gabrielluong — COMMENTED** [review]
My general note is that once this seems good with @cwzilla. We should just convert all the other headers in this PR.

**Comment by sarah541** [comment]
Disabling UI test in this commit : https://github.com/mozilla-mobile/fenix/pull/24171/commits/d39827b8ae0c5d56f7a397a7dffd1e920155d88f

This UI test will be addressed in : https://github.com/mozilla-mobile/fenix/issues/24315

**Pull request closed without merging**